app/services/payment/spam_protection_service.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Payment
+  class SpamProtectionService
+    def initialize(account, payment_status, user_session = nil)
+      @account         = account
+      @payment_status  = payment_status
+      @gateway_setting = account.gateway_setting
+      @user_session    = user_session
+    end
+
+    def call
+      return if payment_successful?
+
+      limiter = ActionLimiter.new(@account)
+      limiter.perform 'invalid_payment'
+    rescue ActionLimiter::ActionLimitsExceededError
+      account.force_suspend!
+      user_session&.revoke!
+    end
+
+    private
+
+    attr_reader :account, :gateway_setting, :payment_status, :user_session
+
+    def payment_successful?
```

Comment:
```suggestion
    alias_method :payment_successful?, :payment_status
```

Reply:
Do we even need this method? in line 13 it could be just `return if payment_status` :thinking:

Reply:
Or even in the intializer `@payment_successful = payment_status` and we make it more understandable

Reply:
Agree with @Martouta.

Reply:
I think it's about readability, but `return if payment_successful` (without the question mark) is acceptable enough.